smallrye-reactive-messaging-jms/src/main/java/io/smallrye/reactive/messaging/jms/JmsSource.java

```diff
@@ -0,0 +1,182 @@
+package io.smallrye.reactive.messaging.jms;
+
+import java.lang.IllegalStateException;
+import java.util.concurrent.Executor;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.AtomicReference;
+
+import javax.jms.*;
+import javax.json.bind.Jsonb;
+
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.reactive.streams.operators.PublisherBuilder;
+import org.eclipse.microprofile.reactive.streams.operators.ReactiveStreams;
+import org.reactivestreams.Publisher;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.reactivex.Flowable;
+import io.reactivex.internal.subscriptions.EmptySubscription;
+import io.reactivex.internal.subscriptions.SubscriptionHelper;
+
+public class JmsSource {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(JmsSource.class);
+    private final PublisherBuilder<ReceivedJmsMessage<?>> source;
+
+    private final JmsPublisher publisher;
+
+    JmsSource(JMSContext context, Config config, Jsonb json, Executor executor) {
+        String name = config.getOptionalValue("destination", String.class)
+                .orElseGet(() -> config.getValue("channel-name", String.class));
+
+        String selector = config.getOptionalValue("selector", String.class).orElse(null);
+        boolean nolocal = config.getOptionalValue("no-local", Boolean.TYPE).orElse(false);
+        boolean broadcast = config.getOptionalValue("broadcast", Boolean.TYPE).orElse(false);
+
+        Destination destination = getDestination(context, name, config);
+
+        JMSConsumer consumer;
+        if (config.getOptionalValue("durable", Boolean.TYPE).orElse(false)) {
+            if (!(destination instanceof Topic)) {
+                throw new IllegalStateException("Invalid destination, only topic can be durable");
```

Comment:
Previously, you throw `IllegalArgumentException` for this condition though. I suggest to stick with `IllegalArgumentException` for the sake of consistence.

Reply:
Good catch, definitely wrong here.

Reply:
Fixed.